### DISCUSSION TITLE
nearest point number

### DISCUSSION THREAD

**Discussion by sinamitnn** [FEALiTE/FEALiTE2D#17] [Ideas]
A function/method for finding the nearest point would be perfect. 
The inputs is going to be the coordinates, and the code is going to give us the number of the nearest point.

**Comment by SaMohamed** [comment]
Hi Sinan,

I try to keep the library consistent to its purpose. So, in the implantation code, anyone can add extension methods or write their own methods to make queries like the one you suggested or queries like the calculating weight of  the structure, etc...

Your suggestion can be implemented easily using this query:

  ```      
 public SortedDictionary<double, Node2D> GetNearestNodestoANode(IEnumerable<Node2D> nodes, Node2D node, double distance)
        {
            var nodesList = new SortedDictionary<double,Node2D>(); 

            foreach (var nNode in structure.Nodes)
            {
                double dist = node.DistanceBetween(nNode);
                if (dist <= distance)
                {
                    nodesList.Add(dist, nNode);
                }
            }

            return nodesList;
        }
```
![image](https://github.com/FEALiTE/FEALiTE2D/assets/21183259/8d77f36e-cf4b-4b61-9d80-96894606fedd)

so in above example, picking a point which have coordinates (0.52, 1.77) and setting the search radius to 0.7, we get a sorted dictionary of points and the nearest point   `var list =   GetNearestNodestoANode(structure.Nodes, new Node2D(0.52, 1.77,""), 0.7);`

you get the following results
![image](https://github.com/FEALiTE/FEALiTE2D/assets/21183259/458d16d4-97fd-4e7b-b1b0-47aa48543c1f)


Regards

**Comment by sinamitnn** [comment]
Hi, 
i understand and follow you. 
It means for me that i have to deep into your code :) 
Thanks also for the query too.
S.
